### ISSUE TITLE
Got websocket message as java.nio.HeapByteBuffer, how to convert to ArrayBuffer type.

### ISSUE THREAD

**Issue by superliqq** [NathanaelA/nativescript-websockets#48]
I have a websocket connection from nativescript app to a backend server which sends byte array through the socket, when message is received, the type it shows is java.nio.HeapByteBuffer,  how to convert them to corresponding javascript type (ArrayBuffer) ? Thanks.

**Comment by edusperoni** [comment]
This is a bug in nativescript-websockets for android.

It should return an ArrayBuffer, but it's returning a java.nio.HeapByteBuffer.

This is my temporary workaround:

```
       var intarray = new Int8Array(message.remaining());
        var i = 0;
        while(message.remaining()) {
            intarray[i] = message.get();
            i++;
        }
        event.data = intarray.buffer;
```

I'll try to search the code for this, but I have absolutely no idea how to fix it.

**Comment by NathanaelA** [comment]
Interesting, it used to return an ArrayBuffer....   Which version of NativeScript are you guys using so that I can test it out and fix it properly.

**Comment by NathanaelA** [comment]
Actually it could be a change in the NS android runtimes; it may now just transfer in a HeapByteButffer where before it would convert it.   I've ran into issues like this before; in fact when I first wrote this library it would crash because of a broken conversion that I had to work around.  :grinning:

**Comment by andy-maca** [comment]
I have the same issue, after check, I think the reason is java_websocket library returns HeapByteBuffer by using ByteBuffer.wrap(...) in https://github.com/TooTallNate/Java-WebSocket/blob/master/src/main/java/org/java_websocket/drafts/Draft_6455.java, I am looking for a way to convert it to ArraBuffer without data copy for performance issue like opengles, the android-runtime has already implemented for direct byte buffer [here](https://github.com/NativeScript/android-runtime/pull/365), but not heap byte buffer.

**Comment by NathanaelA** [comment]
This is now fixed in v1.5.0
